code/game/objects/items/robot/cyborg_gripper.dm

```diff
@@ -0,0 +1,209 @@
+/*   CONTENTS:
+0. generic define gripper
+1. UNIVERSAL GRIPPER
+2. MEDICAL GRIPPER
+3. SERVICE GRIPPER
+4. ENGINEERING GRIPPER
+*/
+
+/obj/item/gripper   // Generic gripper. This should never appear anywhere.
+	name = "generic gripper item"
+	desc = "If you can see this, make an issue report to Github. Something has gone wrong!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gripper"
+	actions_types = list(/datum/action/item_action/drop_gripped_item)
+	var/engineering_machine_interaction = FALSE	// Used to stop non-engi grippers from grabbing cells and lightbulbs from certain machines.
+	var/can_help_up = FALSE	// Used to stop non-medical/service grippers from shaking people awake or helping them up.
+	var/list/can_hold = list(
+		/obj/item
+	)
+
+	var/obj/item/gripped_item    // Item currently being held.
+
+/obj/item/gripper/examine_more(mob/user)
+	. = ..()
+	. += "Cyborg grippers are well-developed, and despite some anatomical differences that manifest in some models, \
+	they can be used just as effectively as a regular hand with enough practice. \
+	Companies like Nanotrasen use software to limit the items that a cyborg can manipulate to a specific pre-defined list, as part of their multi-layered \
+	protections to try and eliminate the chance of a hypothetical synthetic uprising, not wishing to see a repeat of the IPC uprising in 2525."
+
+
+/obj/item/gripper/Initialize(mapload)
+	. = ..()
+	can_hold = typecacheof(can_hold)
+
+/obj/item/gripper/ui_action_click(mob/user)
+	drop_gripped_item()
+
+/obj/item/gripper/proc/drop_gripped_item(silent = FALSE)
+	if(!gripped_item)
+		return
+	if(!silent)
+		to_chat(loc, "<span class='warning'>You drop [gripped_item].</span>")
+	gripped_item.forceMove(get_turf(src))
+	gripped_item = null
+
+/obj/item/gripper/attack_self(mob/user)
+	if(!gripped_item)
+		to_chat(user, "<span class='warning'>[src] is empty.</span>")
+		return
+	gripped_item.attack_self(user)
+
+/obj/item/gripper/attack(mob/living/carbon/M, mob/living/carbon/user)
+	return
+
+/// Grippers are snowflakey so this is needed to to prevent forceMoving grippers after `if(!user.drop_item())` checks done in certain attackby's. // What does this even MEAN - GDN // I don't know either - CRUNCH
```

Comment:
I think I wrote this way back when, and I remember why 😆. If we take engi grippers as an example, they can pick up a battery and place it into an APC. The APC (or whatever object) will do an `if(!user.drop_item()) -> return` check when the borg clicks on the APC with the gripper, which is an anti-drop check. If this code wasn't here, it would drop the gripper to the floor.

Reply:
Fair enough haha!